src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -12559,6 +12557,8 @@ void Unit::RemoveCharmedBy(Unit* charmer)
         }
         player->SetClientControl(this, true);
     }
+    
+    EngageWithTarget(charmer);
```

Comment:
It _should_ still do that. The charm isn't removed until _after_ the evade happens - at this point the mob is in evade mode, and `EngageWithTarget` will fizzle.

I'll make a note to test that, though.